R/block_cv.R

```diff
@@ -0,0 +1,160 @@
+#' Spatial block cross-validation
+#'
+#' Block cross-validation splits the area of your data into a number of
+#' grid cells, or "blocks", and then assigns all data into folds based on the
+#' blocks they fall into. By putting
+#'
+#' @details
+#' The grid blocks are drawn from can be controlled by passing arguments to
+#' [sf::st_make_grid()] via `...`. Some particularly useful arguments include:
+#'
+#' * `cellsize` Target cellsize, expressed as the "diameter" (shortest
+#' straight-line distance between opposing sides; two times the apothem)
+#' of each block, in map units.
+#' * `n` The number of grid blocks in the x and y direction (columns, rows).
+#' * `square` A logical value indicating whether to create square (`TRUE`) or
+#' hexagonal (`FALSE`) cells.
+#'
+#' If both `cellsize` and `n` are provided, then the number of blocks requested
+#' by `n` of sizes specified by `cellsize` will be returned, likely not
+#' lining up with the bounding box of `data`. If only `cellsize`
+#' is provided, then this function will return as many blocks of size
+#' `cellsize` as fit inside the bounding box of `data`. If only `n` is provided,
+#' then `cellsize` will be automatically adjusted to create the requested
+#' number of cells.
+#'
+#' @param data An object of class `sf` or `sfc`.
+#' @param method The method for sampling blocks for cross validation folds.
+#' Currently, only "random" is supported.
+#' @inheritParams rsample::vfold_cv
+#' @param ... Arguments passed to [sf::st_make_grid()].
+#'
+#' @return A tibble with classes `spatial_block_cv`, `rset`, `tbl_df`, `tbl`,
```

Comment:
Should each block assignment method be its own class? I'm not entirely sure how these integrate with the rest of the tidymodels universe (yet!), so I don't know how much actually changes based on this most-granular class name

Reply:
Let's keep this as only `spatial_block_cv` and reevaluate when we add the next approach.